packages/aws-cdk/lib/api/aws-auth/awscli-compatible.ts

```diff
@@ -101,6 +99,25 @@ export class AwsCliCompatible {
       }
     }
 
+    if (!region && isEc2Instance()) {
+      debug('Looking up AWS region in the EC2 Instance Metadata Service (IMDS).');
+      const imdsOptions = {
+        httpOptions: { timeout: 1000, connectTimeout: 1000 }, maxRetries: 2,
+      };
+      const metadataService = new AWS.MetadataService(imdsOptions);
+      const token = await getImdsV2Token(metadataService)
```

Comment:
Kinda don't like the style this is written. It confuses me to combine async/await and thenables in this way.

Why not:

```ts
try {
  const token = await getImdsV2Token(metadataService);
  region = await getRegionFromImds(metadataService, token);
  debug(`Retrieved AWS region "${region}" from the IMDS.`);
} catch (e) {
  debug(`Unable to retrieve region from IMDS: ${e.message}`);
} 
```

Reply:
I was thinking about that, but was concerned about a case where IMDSv2 was not supported -- potentially in partition where it's not implemented. In that situation, `getImdsV2Token` would throw an error but IMDSv1 should still be queried to get the region, just without a token.